### DISCUSSION TITLE
[QnA] 174p 상단 - 맵 리듀스 패턴

### DISCUSSION THREAD

**Discussion by Youngju-Jang** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#50] [CHAPTER 5 스트림 활용 - Q&A]
- 코드만 봤을 경우 맵리듀스 패턴코드보다 count() 코드가 더 간결하고 명확해 보인다.
- 맵 리듀스 패턴이 쉽게 병렬화하는 특징을 가졌다고 하는데,
### count() 보다 맵리듀스 패턴 이 더 효율적인 것일까?

코드 :
```java
int count = 
    menu.steam()
              .map(d -> 1)
              .reduce(0, (a, b) -> a+b);


long count = menu.stream().count();

```

**Comment by corock** [comment]
책에서 소개한 그대로 병렬 처리에 적합할 때 `count()` 대신 맵 리듀스 패턴을 적용하면 될 것 같습니다.
병렬 데이터 처리는 챕터 7에서 곧 나오니 그때 보기로 하고, 애초에 병렬 처리를 효율적으로 이용하려면 데이터 양이 많아야 의미가 있지 않나 싶습니다.

_cf._ 맵 리듀스라는 패턴은 처음 들어보는데, 이게 하둡(_Hadoop_) 구성 요소인 맵 리듀스 의미와 얼추 비슷한 건지 모르겠네요.

**Reply by Irisation23** [reply]
https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=skywood1&logNo=220149255386
